wechatpay/src/main/java/com/stripe/android/payments/wechatpay/WeChatPayAuthActivity.kt

```diff
@@ -3,41 +3,133 @@ package com.stripe.android.payments.wechatpay
 import android.app.Activity
 import android.content.Intent
 import android.os.Bundle
+import android.util.Log
+import androidx.annotation.VisibleForTesting
 import androidx.appcompat.app.AppCompatActivity
 import com.stripe.android.StripeIntentResult
 import com.stripe.android.exception.StripeException
 import com.stripe.android.model.WeChat
 import com.stripe.android.payments.PaymentFlowResult
+import com.tencent.mm.opensdk.modelbase.BaseReq
+import com.tencent.mm.opensdk.modelbase.BaseResp
+import com.tencent.mm.opensdk.modelpay.PayReq
+import com.tencent.mm.opensdk.modelpay.PayResp
+import com.tencent.mm.opensdk.openapi.IWXAPI
+import com.tencent.mm.opensdk.openapi.IWXAPIEventHandler
+import com.tencent.mm.opensdk.openapi.WXAPIFactory
+
+internal class WeChatPayAuthActivity : AppCompatActivity(), IWXAPIEventHandler {
+    @VisibleForTesting
+    internal var weChatApiProvider = { WXAPIFactory.createWXAPI(this, null) }
+
+    private lateinit var weChatApi: IWXAPI
```

Comment:
Can this be `private val weChatApi: IWXAPI by lazy { weChatApiProvider() }`?

Reply:
yes, done